site/zenodo_rdm/subcommunities/request.py

```diff
@@ -113,8 +124,180 @@ class ZenodoSubCommunityRequest(SubCommunityRequest):
 
     available_actions = {
         "delete": actions.DeleteAction,
-        "create": SubcommunityCreateAction,
         "cancel": actions.CancelAction,
+        # Custom implemented actions
+        "create": SubcommunityCreateAction,
         "accept": SubcommunityAcceptAction,
         "decline": DeclineSubcommunity,
     }
+
+
+class SubcommunityInvitationCreateAction(actions.CreateAction):
+    """Represents an accept action used to accept a subcommunity."""
+
+    def execute(self, identity, uow):
+        """Execute approve action."""
+        parent = self.request.created_by.resolve()
+        if not parent.children.allow:
+            raise ValidationError("Assigned parent is not allowed to be a parent.")
+        super().execute(identity, uow)
+
+
+class SubCommunityInvitationAcceptAction(AcceptSubcommunityInvitation):
+    """Represents an accept action used to accept a subcommunity.
+
+    Zenodo re-implementation of the accept action, to also move the records.
+    """
+
+    def _get_community_records(self, community_id):
+        """Get the records of a community."""
+        return current_community_records_service.search(
+            system_identity, community_id=community_id
+        )
+
+    def execute(self, identity, uow):
+        """Execute approve action."""
+        child = self.request.receiver.resolve().id
+        parent = self.request.created_by.resolve().id
+
+        # Move records
+        records = self._get_community_records(child)
+        current_rdm_records.record_communities_service.bulk_add(
+            system_identity, parent, (x["id"] for x in records), uow=uow
+        )
+        super().execute(identity, uow)
+
+
+class SubcommunityInvitationSubmitAction(actions.SubmitAction):
+    """Submit action."""
+
+    def execute(self, identity, uow):
+        """Execute the submit action."""
+        self.request["title"] = "Invitation to join the EU Open Research Repository"
+
+        # example: "May 11, 2024"
+        expires_at = self.request.expires_at.strftime("%B %d, %Y")
+        NAME = "TODO"
+        ACRONYMS = "TODO"
```

Comment:
minor/shelve: these will probably be part of the data/payload of the request, so we can easily pass them as parameters when bulk creating the requests.